private/scribble-util.rkt

```diff
@@ -0,0 +1,24 @@
+#lang racket/base
+
+(module doc racket/base
+
+  (require racket/contract/base
+           scribble/base
+           scribble/core
+           scribble/decode
+           scribble/manual)
+
+  (provide
+    (contract-out
+      [tech/guide
+        ;; Similar to @tech{....}, but creates a link to a `deftech` in The Racket Guide
+        (->* () (#:key (or/c string? #f)) #:rest (listof pre-content?) element?)]
```

Comment:
Could you use `immutable-string?` (from `rebellion/base/immutable-string`) instead of `string?` here? As a rule, I think every function that accepts a string as an input should either only allow immutable strings or make a defensive copy of the input string.